File: leetcode/Easy/0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        memo={}

        def dp(k):
            if k == 1: 
                return 1
            if k == 2: 
                return 2

            if k in memo:
                return memo[k]  

            memo[k] = dp(k-1) + dp(k-2)
            return memo[k]
            
        return dp(n)

File: leetcode/Easy/0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    // if (n === 0 || n === 1) return 1;
    // return climbStairs(n - 1) + climbStairs(n - 2);
    let a=1, b=1
    if (n === 1) return 1;
    if (n === 2) return 2;
    for(let i=0; i<n; i++){
        [a,b] = [b, a+b]
    }
    return a
};